app/helpers/lookup_keys_helper.rb

```diff
@@ -94,25 +88,34 @@ def host_key_with_diagnostic(host, value_hash, key)
     no_value = value.nil? && key.lookup_values.find_by_match("fqdn=#{host.fqdn}")
 
     diagnostic_class = []
-    diagnostic_helper = popover(_("Additional info"), _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => matcher})
+    diagnostic_helper = diagnostic_helper_popover(key, matcher)
     if no_value
       if key.required
         diagnostic_class << 'error'
-        diagnostic_helper = popover(_('No value error'), _("Required parameter without value.<br/><b>Please override!</b> <br><br><b>Description:</b>: %s") % key.description)
+        diagnostic_helper = popover('', _("Required parameter without value.<br/><b>Please override!</b> <br><br><b>Description:</b>: %s") % key.description, :icon => "exclamation-sign")
       else
         diagnostic_class << 'warning'
-        diagnostic_helper = popover(_('No value warning'), _("Optional parameter without value.<br/><i>Won\'t be given to Puppet.</i> <br><br><b>Description:</b> %s") % key.description)
+        diagnostic_helper = popover('', _("Optional parameter without value.<br/><i>Won\'t be given to Puppet.</i> <br><br><b>Description:</b> %s") % key.description, :icon => "warning-sign")
       end
     end
 
-    text_area_class = ['col-md-6']
+    text_area_class = []
     text_area_class << "override-param" if key.overridden?(host)
-    content_tag :div, :class => ['form-group', 'condensed'] + diagnostic_class do
+    value_content_tag(key, text_area_class, original_value, diagnostic_class, diagnostic_helper)
+  end
+
+  def diagnostic_helper_popover(key, matcher)
+    popover('', _("<b>Description:</b> %{desc}<br><b>Type:</b> %{type}<br> <b>Matcher:</b> %{matcher}") % { :desc => key.description, :type => key.key_type, :matcher => matcher})
```

Comment:
I don't agree, this is info about the parameter itself, not just the current value. I think it gives context.